HW0/P4/P4.py

```diff
@@ -40,38 +59,115 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    # Read in observed data from P4_measurements.txt
+    s_obs = np.loadtxt('P4_measurements.txt', delimiter=',')
+    
+    # Create matrix to fix dimensional stretching 
+    stretch_fix = np.array([
+        [(1/rx), 0, 0],
+        [0, (1/ry), 0],
+        [0, 0, (1/rz)]
+        ])
+    
+    # Calculate approximate position
+    s_appr = np.dot(stretch_fix, s_obs.transpose()).transpose()
+    
+    # Plot observed trajectory
+    ax.plot(s_appr[:,0], s_appr[:,1], s_appr[:,2],
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    # Initialize progation matrix
+    A = np.mat([
+        [1, 0, 0, dt, 0, 0],
+        [0, 1, 0, 0, dt, 0],
+        [0, 0, 1, 0, 0, dt],
+        [0, 0, 0, 1-c*dt, 0, 0],
+        [0, 0, 0, 0, 1-c*dt, 0],
+        [0, 0, 0, 0, 0, 1-c*dt]
+        ])
```

Comment:
Thanks for the reply! Apologies for possibly spamming your inbox with notifications- made a few comments along the way and hopefully there’s some setting to turn the emails off.

> On Sep 16, 2015, at 10:43 PM, Gioia Dominedo <notifications@github.com> wrote:
> 
> In HW0/P4/P4.py <https://github.com/harvard-cs205/cs205-homework/pull/170#discussion_r39707485>:
> 
> >  
> >      #####################
> >      # Part 3:
> >      # Use the initial conditions and propagation matrix for prediction
> >      #####################
> >  
> > -    # A = ?
> > -    # a = ?
> > -    # s = ?
> > +    # Initialize progation matrix
> > +    A = np.mat([
> > +        [1, 0, 0, dt, 0, 0],
> > +        [0, 1, 0, 0, dt, 0],
> > +        [0, 0, 1, 0, 0, dt],
> > +        [0, 0, 0, 1-c*dt, 0, 0],
> > +        [0, 0, 0, 0, 1-c*dt, 0],
> > +        [0, 0, 0, 0, 0, 1-c*dt]
> > +        ])
> Thanks! I wasn't familiar with np.diag() but will have to remember that one for future problems.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/harvard-cs205/cs205-homework/pull/170/files#r39707485>.
>